### ISSUE TITLE
App Getting Crash 

### ISSUE THREAD

**Issue by sarankumar291** [zhangao0086/DKImagePickerController#417]
getImageManager().groupDataManager.assetGroupTypes = self.assetGroupTypes

Please Help

**Comment by zhangao0086** [comment]
Hi @sarankumar291  This issue has been fixed in 4.0.0 beta branch, it has not yet been officially released, but you can manually switch to 4.0.0 branch:
```bash
pod 'DKImagePickerController', :git => 'https://github.com/zhangao0086/DKImagePickerController.git', :branch => 'Feature/4.0.0'
```

**Comment by sarankumar291** [comment]
Thanks

**Comment by sarankumar291** [comment]
Use of undeclared type 'DKImagePickerControllerDefaultUIDelegate Getting Error

**Comment by zhangao0086** [comment]
Hi @sarankumar291 `DKImagePickerControllerDefaultUIDelegate` has been renamed to `DKImagePickerControllerBaseUIDelegate`. Please refer to [CustomUIDelegate](https://github.com/zhangao0086/DKImagePickerController/blob/Feature/4.0.0/DKImagePickerControllerDemo/DKImagePickerControllerDemo/CustomUIDelegate/CustomUIDelegate.swift) if you want to customize the UI.

**Comment by sarankumar291** [comment]
Thanks a lot, How to assign selectedAssets Please help

**Comment by sarankumar291** [comment]
How to Export Video Files, writeAVToFile method is not working

**Comment by zhangao0086** [comment]
Please refer to [exporting-to-file](https://github.com/zhangao0086/DKImagePickerController/tree/Feature/4.0.0#exporting-to-file)

**Comment by zhangao0086** [comment]
Closing this thread. Feel free to re-open if your issue is not solved.

**Comment by rasoulmiri** [comment]
in this code crashed
```
  getImageManager().groupDataManager.assetGroupTypes = self.assetGroupTypes
            getImageManager().groupDataManager.assetFetchOptions = self.createAssetFetchOptions()
            getImageManager().groupDataManager.showsEmptyAlbums = self.showsEmptyAlbums
            getImageManager().autoDownloadWhenAssetIsInCloud = self.autoDownloadWhenAssetIsInCloud
```
how fix it for swift 3.2?

**Comment by zhangao0086** [comment]
Hi @rasoulmiri The 4.0.0 version is compatible both Swift 3.2 and Swift 4.
